addon/-private/utils/element/estimate-element-size.js

```diff
@@ -0,0 +1,34 @@
+import { assert } from '@ember/debug';
+
+export default function estimateElementSize(element, fallbackSize, dimension = 'height') {
+  assert(`You called estimateElement height without a fallbackSize`, fallbackSize);
+  assert(`You called estimateElementHeight without an element`, element);
+
+  if (fallbackSize.indexOf('%') !== -1) {
+    return getPercentageSize(element, fallbackSize, dimension);
+  }
+
+  if (fallbackSize.indexOf('em') !== -1) {
+    return getEmSize(element, fallbackSize);
+  }
+
+  return parseInt(fallbackSize, 10);
+}
+
+function getPercentageSize(element, fallbackSize, dimension) {
+  // We use offsetHeight/offsetWidth here to get the element's true size, rather than the
+  // bounding rect which may be scaled with transforms
+  let parentSize = dimension === 'height' ? element.offsetHeight : element.offsetWidth;
+  let percent = parseFloat(fallbackSize);
+
+  return (percent * parentSize / 100.0);
+}
+
+// TODO(kjb) How do we handle this for width? font-size only refers to font height, though we could
+// make a rough estimate of width based on some general-case constant.
```

Comment:
This should be addressed before moving forward with horizontal collection. It's the only thing I'm somewhat hung-up on since there is no definitive answer.

Reply:
font-size applied to width as much as to height

Reply:
Maybe this is a stupid question, but then why are there monospaced fonts? It's my understanding that the width of a font is not directly correlated to its height, but arbitrary.

Reply:
I think `em` and `rem` font-size is a separate concept from actual rendered font size. If you were to say that `em` was based on the current font-size of `12px`, and then to say that this item was `width: 2em`, it would be 24px.